src/ingest_validation_tools/upload.py

```diff
@@ -752,6 +772,30 @@ def __get_no_ref_errors(self) -> dict:
             errors["Files"] = unreferenced_file_paths
         return errors
 
+    def __get_shared_dir_errors(self) -> dict:
+        errors = {}
+        all_non_global_files = self.__get_non_global_files_references()
+        if all_non_global_files:
+            for row_non_global_files, row_references in all_non_global_files.items():
+                row_non_global_files = {
+                    (self.directory_path / "./non_global" / Path(x.strip())): Path(x.strip())
+                    for x in row_non_global_files.split(";")
+                    if x.strip()
+                }
+
+                for full_path_row_non_global_file, rel_path_row_non_global_file in \
+                        row_non_global_files.items():
+                    if not full_path_row_non_global_file.exists():
+                        errors[",".join(row_references)] = \
+                            f"{rel_path_row_non_global_file} not exist in upload."
+        else:
+            # Catch case 2
+            if self.shared_directories:
+                errors["Upload Errors"] = "No non_global_files specified but " \
+                                          "upload has global & non_global directories"
```

Comment:
Very nitpicky: I think this would trigger if an upload had a global _and/or_ non_global directory, rather than needing to have both ("&"). Suggest slight update from `&` -> `and/or`